packages/modeling/src/geometries/poly3/faceNormal.js

```diff
@@ -0,0 +1,39 @@
+const vec3 = require('../../maths/vec3')
+
+/**
+ * Calculate normal vector for a single vertex.
+ */
+const vertexNormal = (vertices, index) => {
```

Comment:
The big performance hit will be the number of vec3.create() calls.

if this function was embedded in the main function then temporarily storage can be allocated for ba and ca, and reused for each call to vertexNormal().

Reply:
Inlined to avoid allocations :heavy_check_mark: